pkg/util/proxy/proxy.go

```diff
@@ -139,7 +139,16 @@ func GetTlsConfigForCluster(ctx context.Context, cluster *clusterapis.Cluster, s
 	}
 	caBundle, err := getClusterCABundle(cluster.Name, caSecret)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get CA bundle for cluster %s: %v", cluster.Name, err)
+		if cluster.Spec.InsecureSkipTLSVerification {
+			return &tls.Config{
+				MinVersion: tls.VersionTLS13,
+				// Ignore false positive warning: "TLS InsecureSkipVerify may be true. (gosec)"
+				// Whether to skip server certificate verification depends on the
+				// configuration(.spec.insecureSkipTLSVerification, defaults to false) in a Cluster object.
+				InsecureSkipVerify: cluster.Spec.InsecureSkipTLSVerification, //nolint:gosec
+			}, nil
+		}
+		return nil, fmt.Errorf("insecureSkipTLSVerification is false, but failed to get CA bundle for cluster %s: %v", cluster.Name, err)
```

Comment:
How about this? @chaosi-zju @CharlesQQ 
```go
func getClusterCABundle(clusterName string, secret *corev1.Secret) []byte {
	caBundle, found := secret.Data[clusterapis.SecretCADataKey]
	if !found {
		return []byte{}
	}
	return caBundle
}

func GetTlsConfigForCluster(ctx context.Context, cluster *clusterapis.Cluster, secretGetter SecretGetterFunc) (*tls.Config, error) {
        ...
        caBundle := getClusterCABundle(cluster.Name, caSecret)
	caCertPool := x509.NewCertPool()
	caCertPool.AppendCertsFromPEM(caBundle)
	return &tls.Config{
		RootCAs:    caCertPool,
		MinVersion: tls.VersionTLS13,
		// Ignore false positive warning: "TLS InsecureSkipVerify may be true. (gosec)"
		// Whether to skip server certificate verification depends on the
		// configuration(.spec.insecureSkipTLSVerification, defaults to false) in a Cluster object.
		InsecureSkipVerify: cluster.Spec.InsecureSkipTLSVerification, //nolint:gosec
	}, nil
```

Reply:
> I think, if the ca in the container is correct, it will still work fine

I got your point.

> How about this?

I think fine

Reply:
Hi, @CharlesQQ 
Can you help update this?

Reply:
> How about this? @chaosi-zju @CharlesQQ
> 
> ```go
> func getClusterCABundle(clusterName string, secret *corev1.Secret) []byte {
> 	caBundle, found := secret.Data[clusterapis.SecretCADataKey]
> 	if !found {
> 		return []byte{}
> 	}
> 	return caBundle
> }
> 
> func GetTlsConfigForCluster(ctx context.Context, cluster *clusterapis.Cluster, secretGetter SecretGetterFunc) (*tls.Config, error) {
>         ...
>         caBundle := getClusterCABundle(cluster.Name, caSecret)
> 	caCertPool := x509.NewCertPool()
> 	caCertPool.AppendCertsFromPEM(caBundle)
> 	return &tls.Config{
> 		RootCAs:    caCertPool,
> 		MinVersion: tls.VersionTLS13,
> 		// Ignore false positive warning: "TLS InsecureSkipVerify may be true. (gosec)"
> 		// Whether to skip server certificate verification depends on the
> 		// configuration(.spec.insecureSkipTLSVerification, defaults to false) in a Cluster object.
> 		InsecureSkipVerify: cluster.Spec.InsecureSkipTLSVerification, //nolint:gosec
> 	}, nil
> ```

motified!!!